File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/HighestBackPriceFunction.java
package cz.fb.manaus.reactor.profit.progress.function;

import cz.fb.manaus.core.model.Side;
import org.springframework.stereotype.Component;

@Component
public class HighestBackPriceFunction extends AbstractPriceReduceFunction {

    public HighestBackPriceFunction() {
        super(Side.BACK, Math::max);
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/HighestBackPriceFunction.kt
package cz.fb.manaus.reactor.profit.progress.function

import cz.fb.manaus.core.model.Side
import org.springframework.stereotype.Component
import kotlin.math.max

@Component
object HighestBackPriceFunction : ProgressFunction by PriceReduceFunction(Side.BACK, { a, b -> max(a, b) })
